querydsl-jpa/README.md

```diff
@@ -22,27 +22,34 @@ And now, configure the Maven APT plugin :
     <plugins>
       ...
       <plugin>
-        <groupId>com.mysema.maven</groupId>
-        <artifactId>apt-maven-plugin</artifactId>
-        <version>1.1.3</version>
-        <executions>
-          <execution>
-            <goals>
-              <goal>process</goal>
-            </goals>
-            <configuration>
-              <outputDirectory>target/generated-sources/java</outputDirectory>
-              <processor>com.querydsl.apt.jpa.JPAAnnotationProcessor</processor>
-            </configuration>
-          </execution>
-        </executions>
-        <dependencies>
-          <dependency>
-            <groupId>com.querydsl</groupId>
-            <artifactId>querydsl-apt</artifactId>
-            <version>${querydsl.version}</version>
-          </dependency>
-        </dependencies>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
+        <configuration>
+            <annotationProcessors>
+                <annotationProcessor>com.querydsl.apt.jpa.JPAAnnotationProcessor</annotationProcessor>
+            </annotationProcessors>
+            <annotationProcessorPaths>
+                <path>
+                    <groupId>com.querydsl</groupId>
+                    <artifactId>querydsl-apt</artifactId>
+                    <version>${querydsl.version}</version>
+                    <classifier>jpa</classifier>
```

Comment:
The classifier here is unnecessairy.

Either you use:

```xml
<dependency>
  <groupId>com.querydsl</groupId>
  <artifactId>querydsl-apt</artifactId>
  <version>${querydsl.version}</version>
  <scope>provided</scope>
  <classifier>jpa</classifier>
</dependency>
```

as _dependency_. Which automatically registers the `com.querydsl.apt.jpa.JPAAnnotationProcessor` as APT processor through a service loader. This means it hooks onto the compiler directly, does not require additional plugins and also just works with compilation from your IDE. This plus its simplicity IMO makes this approach preferred over the plugin, however... 

If you use the plugin, then specifying a classifier is unnecessary there, because you have to provide the `annotationProcessor` class anyway.

Reply:
I see. Thanks for the explanation! 

Do you think we should add both approaches to the docs and let users decide which they prefer? Maybe present one as recommended? Or only document one? Which one? 

As far as I understand the drawback of the approach from this PR is that one would need to specify some extra dependencies depending on the processor they want to use so that the annotations are on the AP classpath. I assume it is not really a problem, since people will normally just copy-paste the snippet from the docs.

While the issue of using querydsl-apt as a provided dependency is that classes from it will appear in the completion. This is probably a far-fetched issue, but some people seem to [find it annoying](https://github.com/querydsl/querydsl/issues/2091#issuecomment-869964558). It surely requires less configuration, but IMO explicit is better than implicit.

We may want to document both as if the user already uses another annotation processor, then they should use the same approach for adding querydsl-apt as it will either disable the other annotation processor or the querydsl processor will do nothing upon adding it. The drawback is that we try to cover a lot of corner cases making documentation harder to follow.

I'm thinking to document the approach from this PR as recommended (explicit better than implicit) and mention the existing annotation processors' caveat pointing out the provided dependency approach as a solution to those. But let me know if you think differently.

Reply:
> Do you think we should add both approaches to the docs and let users decide which they prefer? Maybe present one as recommended? Or only document one? Which one?

Probably best to do both.

1) The approach using the Maven plugin allows for additional configuration to be set
2) The approach using the Maven plugin might be more recognisable for Spring Data MongoDB users, as Spring Data provides an extended Querydsl Annotation Processor for MongoDB.
3) The approach using the Maven plugin might be preferred in specific situations where multiple APT plugins are used and are meant to be executed in a particular order.